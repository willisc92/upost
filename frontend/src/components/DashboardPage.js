import React from "react";
import { connect } from "react-redux";

class DashboardPage extends React.Component {
    render() {
        return (
            <div className="page-header">
                <div className="content-container">
                    {!this.props.isAuthenticated ? (
                        <h1 className="page-header__title">You must login.</h1>
                    ) : (
                        <h1 className="page-header__title">Welcome, ${localStorage.getItem("first_name")}!</h1>
                    )}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.token
    };
};

export default connect(mapStateToProps)(DashboardPage);
