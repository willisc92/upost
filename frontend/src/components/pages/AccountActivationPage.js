import React from "react";
import { startActivation } from "../../actions/activate";

class AccountActivationPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            inProgress: true,
            isValid: undefined
        };
    }

    componentDidMount() {
        const uid = this.props.match.params.uid;
        const token = this.props.match.params.token;
        startActivation(uid, token)
            .then((result) => {
                this.setState(() => {
                    return { inProgress: false, isValid: true };
                });
            })
            .catch((error) => {
                console.log("error", JSON.stringify(error, null, 2));
                this.setState(() => {
                    return { inProgress: false, isValid: false };
                });
            });
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        {this.state.inProgress ? (
                            <h1 className="page-header__title">
                                Your <span>UPost</span> account is currently being activated!
                            </h1>
                        ) : this.state.isValid ? (
                            <h1 className="page-header__title">
                                Thank you for your email confirmation. Please login and complete your registration.
                            </h1>
                        ) : (
                            <h1 className="page-header__title">The activation link is invalid</h1>
                        )}
                    </div>
                </div>
            </div>
        );
    }
}

export default AccountActivationPage;
