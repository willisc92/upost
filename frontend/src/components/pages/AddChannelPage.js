import React from "react";
import ChannelForm from "../forms/ChannelForm";
import { connect } from "react-redux";
import { addChannel } from "../../actions/channels";

class AddChannelPage extends React.Component {
    constructor(props) {
        super(props);
    }

    onSubmit = (channel) => {
        this.props
            .addChannel({ ...channel })
            .then(() => this.props.history.push("/myChannels"))
            .catch(() => {});
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add a Channel</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ChannelForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addChannel: (channel) => dispatch(addChannel(channel))
});

export default connect(
    null,
    mapDispatchToProps
)(AddChannelPage);
