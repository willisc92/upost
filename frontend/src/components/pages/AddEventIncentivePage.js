import React from "react";
import { addIncentivePackage } from "../../actions/incentivePackage";
import { getCurrentUser } from "../../actions/auth";
import IncentiveForm from "../forms/IncentiveForm";
import { connect } from "react-redux";
import { startSetEvent, clearEvents } from "../../actions/events";

import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

class AddEventIncentivePage extends React.Component {
    constructor(props) {
        super(props);
    }

    componentWillMount() {
        this.props.clearEvents();
        const event_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startSetEvent(event_id)
                    .then((event_res) => {
                        if (res.data.username !== event_res.data.event_owner) {
                            this.props.history.push(`/myChannels`);
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    onSubmit = (incentive) => {
        const post_id = this.props.event.post;
        this.props
            .addIncentivePackage(incentive)
            .then((res) => this.props.history.push(`/post-events/${post_id}`))
            .catch((err) => {
                console.log(err);
            });
    };

    goBack = () => {
        const event_id = this.props.match.params.id;
        const post_id = this.props.event.post;
        this.props.history.push(`/myPosts/${post_id}/events/${event_id}/edit`);
    };

    goToIncentive = () => {
        const event_id = this.props.match.params.id;
        this.props.history.push(`/myEvents/${event_id}/editIncentive`);
    };

    render() {
        const read_only = !!this.props.event && this.props.event.deleted_flag;
        const existing_incentive = !!this.props.event && !!this.props.event.event_incentive;

        return (
            !!this.props.event && (
                <div>
                    <Box bgcolor="secondary.main" py={3}>
                        <Container fixed>
                            <Typography gutterBottom variant="h2">
                                Add an Incentive Package to Event:{" "}
                                <Typography variant="inherit" display="inline" color="primary">
                                    {this.props.event && this.props.event.event_title}
                                </Typography>
                            </Typography>

                            {existing_incentive ? (
                                <Box paddingBottom={2}>
                                    <Typography variant="h2" color="error" gutterBottom>
                                        There is already an existing incentive tied to this event.
                                    </Typography>
                                    <Button variant="contained" color="primary" onClick={this.goToIncentive}>
                                        Go to Incentive
                                    </Button>
                                </Box>
                            ) : (
                                read_only && (
                                    <Typography variant="h2" color="error" gutterBottom>
                                        The event this incentive will be tied to is deleted. Restore it before adding an
                                        incentive.
                                    </Typography>
                                )
                            )}
                            <Box>
                                <Button variant="contained" color="primary" onClick={this.goBack}>
                                    Go to Event
                                </Button>
                            </Box>
                        </Container>
                    </Box>
                    <Container fixed>
                        <IncentiveForm
                            onSubmit={this.onSubmit}
                            nextStep="Save"
                            event={this.props.event}
                            fromEvent={true}
                            read_only={read_only || existing_incentive}
                        />
                    </Container>
                </div>
            )
        );
    }
}

const mapStateToProps = (state) => ({
    event: state.events.events.length !== 0 ? state.events.events : null
});

const mapDispatchToProps = (dispatch) => ({
    clearEvents: () => dispatch(clearEvents()),
    addIncentivePackage: (incentive) => dispatch(addIncentivePackage(incentive)),
    startSetEvent: (id) => dispatch(startSetEvent(id))
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AddEventIncentivePage);
