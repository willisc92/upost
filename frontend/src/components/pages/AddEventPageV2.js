import React from "react";
import { addEvent } from "../../actions/events";
import { getCurrentUser } from "../../actions/auth";
import EventForm from "../forms/EventForm";
import IncentiveForm from "../forms/IncentiveForm";
import { connect } from "react-redux";
import { startGetPost, clearPosts } from "../../actions/posts";
import { addIncentivePackage } from "../../actions/incentivePackage";
import { RecurringModal } from "../modals/RecurringModal";
import { mapFrequencyToRRule, mapDayToRRule } from "../../utils/recurring";
import { RRule } from "rrule";

// TODO:  Refactor this to be an add Event+Incentive Page, with the option for recurring submit from AddPostPage.
// TODO:  Event and Incentive Forms should save base event/incentive locally to the state, and wait for a final submit.
// TODO:  Create toggle to render IncentiveForm/unrender Event Form.
// TODO:  Add a Save All button that conditionally submits an event, event+incentive, or recurring event+recurring incentive.

class AddEventPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            step: "Event",
            base_event: null,
            base_incentive: null,
            finished: false,
            recurringOpen: false,
            rrule_payload: {
                recurringFrequency: "none",
                endSelection: "none",
                numOccurences: 0,
                endDate: null,
                byweekday: [
                    { id: 0, value: "0", label: "Monday", isChecked: false },
                    { id: 1, value: "1", label: "Tuesday", isChecked: false },
                    { id: 2, value: "2", label: "Wednesday", isChecked: false },
                    { id: 3, value: "3", label: "Thursday", isChecked: false },
                    { id: 4, value: "4", label: "Friday", isChecked: false },
                    { id: 5, value: "5", label: "Saturday", isChecked: false },
                    { id: 6, value: "6", label: "Sunday", isChecked: false }
                ]
            },
            rrule_event_starts: null,
            rrule_event_ends: null,
            rrule_incentive_starts: null,
            rrule_incentive_ends: null,
            error: ""
        };
    }

    componentWillMount() {
        this.props.clearPosts();
        const post_id = this.props.match.params.id;
        getCurrentUser()
            .then((res) => {
                this.props
                    .startGetPost(post_id)
                    .then((post_res) => {
                        if (res.data.username !== post_res.data[0].user) {
                            this.props.history.push(`/myChannels`);
                        }
                    })
                    .catch((err) => {
                        console.log(JSON.stringify(err, null, 2));
                    });
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    }

    toggleForms = () => {
        if (this.state.step === "Event") {
            this.setState({
                step: "Incentive"
            });
        }

        if (this.state.step === "Incentive") {
            this.setState({
                step: "Event"
            });
        }
    };

    goBack = () => {
        const post_id = this.props.match.params.id;
        this.props.history.push(`/myPosts/${post_id}/events`);
    };

    recurringSubmit = async (payload) => {
        await this.setState(() => ({
            rrule_payload: payload,
            recurringOpen: false
        }));

        this.generateEventRule(payload);
        this.generateIncentiveRule(payload);
    };

    recurringOpen = () => {
        this.setState(() => ({
            recurringOpen: true
        }));
    };

    clearBaseIncentive = () => {
        this.setState(() => ({
            base_incentive: null,
            step: "Event",
            rrule_incentive_starts: null,
            rrule_incentive_ends: null
        }));
    };

    generateEventRule = (payload) => {
        if (payload.recurringFrequency === "none") {
            this.setState(() => ({
                rrule_event_starts: null,
                rrule_event_ends: null
            }));
        }

        if (!!this.state.base_event && payload.endSelection === "numOccurence") {
            this.setState(() => ({
                event_error: "",
                rrule_event_starts: new RRule({
                    freq: mapFrequencyToRRule(payload.recurringFrequency),
                    dtstart: this.state.base_event.planned_start_date,
                    count: payload.numOccurences
                }),
                rrule_event_ends: new RRule({
                    freq: mapFrequencyToRRule(payload.recurringFrequency),
                    dtstart: this.state.base_event.planned_end_date,
                    count: payload.numOccurences
                })
            }));
        }

        if (!!this.state.base_event && payload.endSelection === "onDate") {
            if (this.state.base_event.planned_end_date > payload.endDate) {
                this.setState(() => ({
                    event_error:
                        "Error, cannot have a recurring end date before the event end date.  Change either the recurring end date or the end date of the event."
                }));
                return null;
            } else {
                this.setState(() => ({
                    event_error: "",
                    rrule_event_starts: new RRule({
                        freq: mapFrequencyToRRule(payload.recurringFrequency),
                        dtstart: this.state.base_event.planned_start_date,
                        until: payload.endDate,
                        byweekday: payload.byweekday.map((day) => {
                            return mapDayToRRule(day);
                        })
                    }),
                    rrule_event_ends: new RRule({
                        freq: mapFrequencyToRRule(payload.recurringFrequency),
                        dtstart: this.state.base_event.planned_end_date,
                        until: payload.endDate,
                        byweekday: payload.byweekday.map((day) => {
                            return mapDayToRRule(day);
                        })
                    })
                }));
            }
        }
    };

    generateIncentiveRule = (payload) => {
        if (!!this.state.base_incentive && payload.recurringFrequency !== "none") {
            this.setState(() => ({
                rrule_incentive_starts: null,
                rrule_incentive_ends: null
            }));
        }

        if (!!this.state.base_incentive && payload.endSelection === "numOccurence") {
            this.setState(() => ({
                incentive_error: "",
                rrule_incentive_starts: new RRule({
                    freq: mapFrequencyToRRule(payload.recurringFrequency),
                    dtstart: this.state.base_incentive.planned_start_date,
                    count: payload.numOccurences
                }),
                rrule_incentive_ends: new RRule({
                    freq: mapFrequencyToRRule(payload.recurringFrequency),
                    dtstart: this.state.base_incentive.planned_end_date,
                    count: payload.numOccurences
                })
            }));
        }

        if (!!this.state.base_incentive && payload.endSelection === "onDate") {
            if (this.state.base_incentive.planned_end_date > payload.endDate) {
                this.setState(() => ({
                    incentive_error:
                        "Error, cannot have a recurring end date before the incentive end date.  Change either the recurring end date or the end date of the incentive."
                }));
                return null;
            } else {
                this.setState(() => ({
                    incentive_error: "",
                    rrule_incentive_starts: new RRule({
                        freq: mapFrequencyToRRule(payload.recurringFrequency),
                        dtstart: this.state.base_incentive.planned_start_date,
                        until: payload.endDate,
                        byweekday: payload.byweekday.map((day) => {
                            return mapDayToRRule(day);
                        })
                    }),
                    rrule_incentive_ends: new RRule({
                        freq: mapFrequencyToRRule(payload.recurringFrequency),
                        dtstart: this.state.base_incentive.planned_end_date,
                        until: payload.endDate,
                        byweekday: payload.byweekday.map((day) => {
                            return mapDayToRRule(day);
                        })
                    })
                }));
            }
        }
    };

    submitAll = async () => {
        await this.setState(() => ({
            finished: true
        }));

        this.submitButtonRef.click();
    };

    mapRRulesToText = () => {
        let rrule_event_string = null;
        let rrule_incentive_string = null;
        if (!!this.state.rrule_event_starts) {
            rrule_event_string = this.state.rrule_event_starts.toText();
        }
        if (!!this.state.rrule_incentive_starts) {
            rrule_incentive_string = this.state.rrule_incentive_starts.toText();
        }

        if (!!rrule_event_string && !rrule_incentive_string) {
            return "Events will reoccur ".concat(rrule_event_string);
        }

        if (!!rrule_event_string && !!rrule_incentive_string && rrule_event_string === rrule_incentive_string) {
            return "Events and incentives will reoccur ".concat(rrule_event_string);
        }

        if (!!rrule_event_string && !!rrule_incentive_string && rrule_event_string !== rrule_incentive_string) {
            return "Events will reoccur "
                .concat(rrule_event_string)
                .concat(".  Incentives will reoccur ")
                .concat(rrule_incentive_string);
        }

        return null;
    };

    handleRecurringEventSubmit = (data) => {
        const promises = [];

        for (var i = 0; i < this.state.rrule_event_starts.all().length; i++) {
            const new_payload = {
                ...data,
                planned_start_date: this.state.rrule_event_starts.all()[i],
                planned_end_date: this.state.rrule_event_ends.all()[i]
            };

            promises.push(this.props.addEvent(new_payload));
        }

        Promise.all(promises)
            .then(() => {
                this.goBack();
            })
            .catch((err) => {
                console.log(JSON.stringify(err, null, 2));
            });
    };

    onSubmit = async (data) => {
        if (this.state.step === "Event") {
            await this.setState(() => ({
                base_event: data
            }));
            this.generateEventRule(this.state.rrule_payload);
            if (this.state.finished === false) {
                alert("Event Draft Saved");
            }
        } else if (this.state.step === "Incentive") {
            await this.setState(() => ({
                base_incentive: data
            }));

            this.generateIncentiveRule(this.state.rrule_payload);
            if (this.state.finished === false) {
                alert("Incentive Draft Saved");
            }
        }

        if (!!this.state.finished) {
            if (!this.state.base_event) {
                await this.setState(() => ({
                    finished: false,
                    error: "Must add an event prior to submission"
                }));
                return;
            }

            // Handle Single Event Submit
            if (!this.state.rrule_event_starts) {
                this.props
                    .addEvent(this.state.base_event)
                    .then((result) => {
                        console.log(result);
                        if (!!this.state.base_incentive) {
                            let payload = {
                                ...this.state.base_incentive,
                                event: result.data.event_id
                            };
                            this.props
                                .addIncentivePackage(payload)
                                .then((result) => {
                                    this.goBack();
                                })
                                .catch((err) => {
                                    console.log(JSON.stringify(err, null, 2));
                                });
                        } else {
                            this.goBack();
                        }
                    })
                    .catch((err) => {
                        this.setState(() => ({
                            finished: false
                        }));
                        console.log(JSON.stringify(err, null, 2));
                    });
                return;
            }

            // Handle Recurring Event Submit w/o Incentives
            if (!!this.state.rrule_event_starts && !this.state.rrule_incentive_starts) {
                this.handleRecurringEventSubmit(this.state.base_event);
            }
        }

        // Handle:
        // Recurring Event Submit
        // Recurring Incentive Submit
        // If any error raised - set state.finished === false and log error.

        // const post_id = this.props.match.params.id;
        // this.props
        //     .addEvent(data)
        //     .then((res) => this.props.history.push(`/myPosts/${post_id}/events`))
        //     .catch((err) => {
        //         console.log(JSON.stringify(err, null, 2));
        //     });
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">
                            Add an Event to Post: <span>{this.props.post && this.props.post.post_title}</span>
                        </h1>
                        <div className="page-header__actions">
                            <span>
                                <button className="button" onClick={this.goBack}>
                                    See All Post Events
                                </button>{" "}
                                <button className="button" onClick={this.recurringOpen}>
                                    Open Recurring Settings
                                </button>{" "}
                                {!!this.state.base_incentive && (
                                    <button className="button" onClick={this.clearBaseIncentive}>
                                        Clear Incentive
                                    </button>
                                )}
                            </span>
                        </div>
                    </div>
                </div>
                <div className="content-container">
                    <RecurringModal isOpen={this.state.recurringOpen} onSubmit={this.recurringSubmit} />
                    <div className="input_group__item">
                        <button className="button" onClick={this.toggleForms}>
                            Open {this.state.step === "Event" ? "Incentive" : "Event"} Settings
                        </button>
                    </div>
                    <div className="input_group__item">
                        <h3>{this.mapRRulesToText()}</h3>
                    </div>
                    <div className="input_group__item">
                        {!!this.state.event_error && <p className="form__error">{this.state.event_error}</p>}
                        {!!this.state.incentive_error && <p className="form__error">{this.state.incentive_error}</p>}
                        {!!this.state.error && <p className="form__error">{this.state.error}</p>}
                    </div>
                    {this.state.step === "Event" && (
                        <div className="input_group__item">
                            <EventForm
                                event={this.state.base_event}
                                read_only={false}
                                id="Event"
                                post={this.props.match.params.id}
                                description={!!this.props.post ? this.props.post.post_description : ""}
                                onSubmit={this.onSubmit}
                                nextStep="Save Event Draft"
                            />
                        </div>
                    )}
                    {this.state.step === "Incentive" && (
                        <div className="input_group__item">
                            <IncentiveForm
                                id="Incentive"
                                description={!!this.props.post ? this.props.post.post_description : ""}
                                incentivePackage={this.state.base_incentive}
                                onSubmit={this.onSubmit}
                                nextStep="Save Incentive Draft"
                                event={this.state.base_event}
                                fromEvent={true}
                            />
                        </div>
                    )}

                    <button className="button" onClick={this.submitAll}>
                        Submit All
                    </button>

                    <button
                        className="button__invisible"
                        type="submit"
                        form={this.state.step}
                        ref={(node) => {
                            this.submitButtonRef = node;
                        }}
                    />
                </div>
            </div>
        );
    }
}

const mapStateToprops = (state) => ({
    post: state.posts.posts[0]
});

const mapDispatchToProps = (dispatch) => ({
    clearPosts: () => dispatch(clearPosts()),
    addEvent: (event) => dispatch(addEvent(event)),
    addIncentivePackage: (incentive) => dispatch(addIncentivePackage(incentive)),
    startGetPost: (id) => dispatch(startGetPost(id))
});

export default connect(
    mapStateToprops,
    mapDispatchToProps
)(AddEventPage);
