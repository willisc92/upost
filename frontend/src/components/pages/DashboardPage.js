import React from "react";
import { connect } from "react-redux";
import { getAllInterests, startSetUserInterests } from "../../actions/interests";
import { startSetInterestRandomPosts } from "../../actions/posts";
import MyPostSummary from "../MyPostSummary";

export class DashboardPage extends React.Component {
    componentDidMount() {
        if (this.props.isAuthenticated) {
            this.props.startSetUserInterests().then(() => {
                this.props.startSetInterestRandomPosts(this.props.interests);
            });
        } else {
            this.props.getAllInterests().then(() => {
                this.props.startSetInterestRandomPosts(this.props.interests);
            });
        }
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        {!this.props.isAuthenticated ? (
                            <h1 className="page-header__title">You must login.</h1>
                        ) : (
                            <h1 className="page-header__title">{`Welcome, ${localStorage.getItem("first_name")}!`}</h1>
                        )}
                    </div>
                </div>
                <div className="content-container">
                    {this.props.interestRandomPosts.map((interestPosts) => {
                        return (
                            <div key={interestPosts.tag}>
                                <h1>{interestPosts.tag}</h1>
<<<<<<< HEAD
                                {/* <div className="list-parent"> */}
                                {interestPosts.posts.map((post) => {
                                    return (
                                        // <div key={post.post_id} className="list-box">
                                        <MyPostSummary post={post} pathName={`/post/${post.post_id}`} />
                                        // </div>
=======
                                {interestPosts.posts.map((post) => {
                                    return (
                                        <MyPostSummary
                                            key={post.post_id}
                                            post={post}
                                            pathName={`/post/${post.post_id}`}
                                        />
>>>>>>> react-pages
                                    );
                                })}
                            </div>
                            // </div>
                        );
                    })}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.token,
        interests: state.userInterests.userInterests,
        interestRandomPosts: state.posts.interestRandomPosts
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        getAllInterests: () => dispatch(getAllInterests()),
        startSetInterestRandomPosts: (interests) => dispatch(startSetInterestRandomPosts(interests)),
        startSetUserInterests: () => dispatch(startSetUserInterests())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DashboardPage);
