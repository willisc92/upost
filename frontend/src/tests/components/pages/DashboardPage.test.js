import React from "react";
import { shallow } from "enzyme";
import { DashboardPage } from "../../../components/pages/DashboardPage";

test("Should render Dashboard page for non logged in user", () => {
    const isAuthenticated = false;
    const wrapper = shallow(<DashboardPage isAuthenticated={isAuthenticated} />);
    expect(wrapper).toMatchSnapshot();
});

test("Should render Dashboard page for logged in user", () => {
    const isAuthenticated = true;
    localStorage.setItem("first_name", "WILLIS");
    const wrapper = shallow(<DashboardPage isAuthenticated={isAuthenticated} />);
    expect(wrapper).toMatchSnapshot();
});
