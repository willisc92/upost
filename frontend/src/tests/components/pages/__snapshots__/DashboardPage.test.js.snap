// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Dashboard page for logged in user 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Welcome, WILLIS!
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      key="Interest 1"
    >
      <h1>
        Interest 1
      </h1>
      <div
        className="list-parent"
      >
        <div
          className="list-box"
          key="1"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 2,
                "cost": 0,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 1,
                  "community": "University of Calgary",
                  "location": "test",
                  "planned_end_date": "2019-06-05T16:10:53-06:00",
                  "planned_start_date": "2019-06-04T16:10:51-06:00",
                  "post": 1,
                },
                "post_id": 1,
                "post_timestamp": "2019-06-17T15:43:48.090766-06:00",
                "post_title": "Post 1",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="2"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 2,
                "cost": 1,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 100,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-02-01T01:00:00-07:00",
                  "planned_start_date": "2019-01-01T01:00:00-07:00",
                  "post": 2,
                },
                "post_id": 2,
                "post_timestamp": "2019-06-17T16:13:02.775479-06:00",
                "post_title": "Post 2",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="3"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 123,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 123,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-01-01T01:00:00-07:00",
                  "planned_start_date": "2019-01-01T01:00:00-07:00",
                  "post": 3,
                },
                "post_id": 3,
                "post_timestamp": "2019-06-17T16:26:49.774048-06:00",
                "post_title": "Post 3",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="4"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 12300,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 10,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-06-17T17:33:01.112000-06:00",
                  "planned_start_date": "2019-06-17T16:33:01.112000-06:00",
                  "post": 4,
                },
                "post_id": 4,
                "post_timestamp": "2019-06-17T16:33:01.093135-06:00",
                "post_title": "Post 4",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="5"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 0,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": null,
                "post_id": 5,
                "post_timestamp": "2019-06-17T16:34:27.423314-06:00",
                "post_title": "Post 5",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="6"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 12300,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 0,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-06-17T17:35:28.118000-06:00",
                  "planned_start_date": "2019-06-17T16:35:28.118000-06:00",
                  "post": 6,
                },
                "post_id": 6,
                "post_timestamp": "2019-06-17T16:35:28.100463-06:00",
                "post_title": "Post 6",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should render Dashboard page for non logged in user 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        You must login.
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <div
      key="Interest 1"
    >
      <h1>
        Interest 1
      </h1>
      <div
        className="list-parent"
      >
        <div
          className="list-box"
          key="1"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 2,
                "cost": 0,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 1,
                  "community": "University of Calgary",
                  "location": "test",
                  "planned_end_date": "2019-06-05T16:10:53-06:00",
                  "planned_start_date": "2019-06-04T16:10:51-06:00",
                  "post": 1,
                },
                "post_id": 1,
                "post_timestamp": "2019-06-17T15:43:48.090766-06:00",
                "post_title": "Post 1",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="2"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 2,
                "cost": 1,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 100,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-02-01T01:00:00-07:00",
                  "planned_start_date": "2019-01-01T01:00:00-07:00",
                  "post": 2,
                },
                "post_id": 2,
                "post_timestamp": "2019-06-17T16:13:02.775479-06:00",
                "post_title": "Post 2",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="3"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 123,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 123,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-01-01T01:00:00-07:00",
                  "planned_start_date": "2019-01-01T01:00:00-07:00",
                  "post": 3,
                },
                "post_id": 3,
                "post_timestamp": "2019-06-17T16:26:49.774048-06:00",
                "post_title": "Post 3",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="4"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 12300,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 10,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-06-17T17:33:01.112000-06:00",
                  "planned_start_date": "2019-06-17T16:33:01.112000-06:00",
                  "post": 4,
                },
                "post_id": 4,
                "post_timestamp": "2019-06-17T16:33:01.093135-06:00",
                "post_title": "Post 4",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="5"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 0,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": null,
                "post_id": 5,
                "post_timestamp": "2019-06-17T16:34:27.423314-06:00",
                "post_title": "Post 5",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
        <div
          className="list-box"
          key="6"
        >
          <MyPostSummary
            post={
              Object {
                "channel": 1,
                "cost": 12300,
                "deleted_flag": false,
                "email": "willisc92@hotmail.com",
                "phone_number": "4034010115",
                "picture": null,
                "post_description": "Test",
                "post_event": Object {
                  "capacity": 0,
                  "community": "University of Calgary",
                  "location": "ICT 122",
                  "planned_end_date": "2019-06-17T17:35:28.118000-06:00",
                  "planned_start_date": "2019-06-17T16:35:28.118000-06:00",
                  "post": 6,
                },
                "post_id": 6,
                "post_timestamp": "2019-06-17T16:35:28.100463-06:00",
                "post_title": "Post 6",
                "poster_name": "Willis",
                "tags": Array [
                  "Interest 1",
                ],
                "user": "willisc92",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
